### PR TITLE
feat(wallet): new browser wallet connection model

### LINKED ISSUE #5443: Support new connection model for Browser Wallet API
## The Chore

In addition to requirement covered in #4717 the Browser wallet will soon emit an event to say that iit has been disconnected. 

Apps need to respond to this properly.

## Tasks

- [x] Change `InjectedConnector` to support new model
- [x] Add event handling to the `InjectedConnector`
- [x] Notify user that the wallet has been disconnected (Console)
- [x] Notify user that the wallet has been disconnected (Governance)
- [x] Tweak the `is alive timeout` (to 1s) so it doesn't appear as disconnected when a bit lagged behind (happened on gov 🤷 )
- [ ] Test

### PR DESCRIPTION
**Pull request by asiaznik**
# Related issues 🔗

Closes #5443

# Description ℹ️

This PR adds support for the recent changes in Vega browser wallet:
* `client.disconnected` event
* passing `chainId` for multi network wallet capabilities (instead of single network wallets)

# Demo 📺

https://github.com/vegaprotocol/frontend-monorepo/assets/1980305/83902e8d-025a-41ad-a22c-9f73ddbb389a

# Technical 👨‍🔧

Slightly refactored how chain id for connection is being taken - it is not a part of the wallet context.

### CODE REVIEW

File: libs/wallet/src/connectors/injected-connector.ts [hunk 1]
```diff
@@ -48,14 +55,34 @@ export const InjectedConnectorErrors = {
...
+      this.isConnected = true;
+    } catch {
+      throw new Error(
+        `could not connect to the vega wallet on chain: ${chainId}`
+      );
+    }
+  }
+
+  onDisconnect(cb: () => void) {
+    window.vega.on('disconnect', () => {
```
Comment (dexturr) [inline libs/wallet/src/connectors/injected-connector.ts:82]: I think this is the event name 😄 

```suggestion
    window.vega.on('client.disconnect', () => {
```

Comment (dexturr) [reply libs/wallet/src/connectors/injected-connector.ts:82]: https://github.com/vegaprotocol/vegawallet-browser/pull/667

Comment (asiaznik) [reply libs/wallet/src/connectors/injected-connector.ts]: it's `client.disconnected` :D

**Review by mattrussell36 — APPROVED** [review]

**Review by daro-maj — CHANGES_REQUESTED** [review]
Add a toast after disconnect wallet in governance.

**Review by daro-maj — APPROVED** [review]

**Review by MadalinaRaicu — APPROVED** [review]

**Pull request merged**